src/utils/style-sheet.ts

```diff
@@ -30,21 +30,45 @@ function makeStyleTag() {
   return tag
 }
 
-export default function StyleSheet({
-  speedy = !isDev && !isTest,
-  maxLength = 65000
-} = {}) {
-  this.isSpeedy = speedy // The big drawback here is that the css won't be editable in devtools
-  this.sheet = undefined
-  this.tags = []
-  this.maxLength = maxLength
-  this.ctr = 0
+type Writeable<T> = {
+  -readonly [P in keyof T]: T[P]
```

Comment:
Why is there a prefix `-` for readonly? I didn't find any documentation around this in https://www.typescriptlang.org/docs/handbook/advanced-types.html

Reply:
It's an attribute _removal_. I'm removing `readonly` attributes from this object, since its props are otherwise sealed as `readonly`

Reply:
https://github.com/Microsoft/TypeScript/wiki/What%27s-new-in-TypeScript#improved-control-over-mapped-type-modifiers

Reply:
oh nice! thanks for the docs! 😍

Reply:
This is some magic stuff @mshwery 😄